### ISSUE TITLE
Why is it necessary to anneal the clip coefficient in PPO?

### ISSUE THREAD

**Issue by liruiluo** [DLR-RM/stable-baselines3#1763]
### ❓ Question

I have read many related articles about implementing PPO, but no article mentioned that the annealing of the clip coefficient of PPO plays a key role. However, in this library, this function is implemented and its importance is not emphasized. So, has anyone tested and found that this feature increases the rewards of most games?

### Checklist

- [X] I have checked that there is no similar [issue](https://github.com/DLR-RM/stable-baselines3/issues) in the repo
- [X] I have read the [documentation](https://stable-baselines3.readthedocs.io/en/master/)
- [X] If code there is, it is [minimal and working](https://github.com/DLR-RM/stable-baselines3/issues/982#issuecomment-1197044014)
- [X] If code there is, it is formatted using the [markdown code blocks](https://help.github.com/en/articles/creating-and-highlighting-code-blocks) for both code and stack traces.

**Comment by araffin** [comment]
Hello,
i suppose you have an understanding of this coefficient does? and what it implies to decrease it over time?

you can check out the rl zoo, try to remove the annealing on environments where it is used and see what happens vs what you expect (on cartpole for instance : https://github.com/DLR-RM/rl-baselines3-zoo/blob/28dc22896abceb68ea5258dd7dee1d20a691864a/hyperparams/ppo.yml#L32)

**Comment by liruiluo** [comment]
> Hello, i suppose you have an understanding of what this coefficient does? and what it implies to decrease it over time?
> 
> you can check out the rl zoo, try to remove the annealing on environments where it is used and see what happens vs what you expect (on cartpole for instance : https://github.com/DLR-RM/rl-baselines3-zoo/blob/28dc22896abceb68ea5258dd7dee1d20a691864a/hyperparams/ppo.yml#L32)

I understand that there will be oscillations without attenuation, but if attenuation is added, won't it make it impossible to learn effectively in the later stages of training? (This is what I don't understand)

**Comment by araffin** [comment]
> won't it make it impossible to learn effectively in the later stages of training?

It will have a similar effect to reducing the learning rate, changes made to the policy will be smaller and smaller.

**Comment by liruiluo** [comment]
> > won't it make it impossible to learn effectively in the later stages of training?
> 
> It will have a similar effect to reducing the learning rate, changes made to the policy will be smaller and smaller.

Another question is, does lin_0.1 refer to the decay from 0.1 to 0 or from 1 to 0.1? I can’t seem to find this callback in the library)

**Comment by araffin** [comment]
> I can’t seem to find this callback in the library)

it is defined in the RL Zoo and it is from 0.1 to 0:
https://github.com/DLR-RM/rl-baselines3-zoo/blob/28dc22896abceb68ea5258dd7dee1d20a691864a/rl_zoo3/exp_manager.py#L335-L338C42

and

https://github.com/DLR-RM/rl-baselines3-zoo/blob/28dc22896abceb68ea5258dd7dee1d20a691864a/rl_zoo3/utils.py#L280
